app/models/guide.rb

```diff
@@ -104,7 +110,36 @@ def compatibility_label
     end
   end
 
-  private
+protected
+
+  # TODO: Fairly simplistic. This should be expanded to somehow take into
+  # consideration stages and selected practices
+  def calculate_completeness_score
+    total = 0.0
+    counted = 0.0
+    fields.keys.each do |key|
+      total += 1
+      if read_attribute(key)
+        counted += 1
+      end
+    end
+
+    write_attributes(completeness_score: counted / total)
+  end
+
+  # TODO: Fairly simplistic. Should probably be normalized properly.
+  # Right now normalization is based on the highest impressions, and how
+  # this one stacks up. It should probably also take into consideration
+  # How many gardens this thing is in.
+  def calculate_popularity_score
+    top_guides = Guide.all.sort_by{|guide| guide[:impressions_field]}.reverse
```

Comment:
Space missing to the left of {.<br>Space between { and | missing.<br>Space missing inside }.